gobo/awesome/sound.lua

```diff
@@ -184,15 +186,19 @@ function sound.new()
          local x = mouse.screen.geometry.width - 800
          local y = 24
          local killed = false
-         for c in awful.client.iterate(function (c) return c.name == "ncpamixer" end, nil, mouse.screen) do
+         for c in awful.client.iterate(function (c) return c.name == ""..mixer.."" end, nil, mouse.screen) do
             c:kill()
             killed = true
          end
          if not killed then
-            awful.util.spawn("urxvt -geometry 100x20+"..x.."+"..y.." -cr green -title ncpamixer -fn '*-lode sans mono-*' -fb '*-lode sans mono-*' -fi '*-lode sans mono-*' -fbi '*-lode sans mono-*' -depth 32 --color0 rgba:2F00/3F00/3F00/e000 -bg rgba:2F00/3F00/3F00/e000 --color4 '#2F3F3F' --color6 '#8aa' --color11 '#2ee' --color14 '#acc' -b 0 +sb -e ncpamixer") -- or whatever your preferred sound mixer is
+					 if (terminal == "urxvt") then
```

Comment:
is this variable `terminal` defined anywhere?

Reply:
on rc.lua
```lua
-- This is used later as the default terminal and editor to run.
terminal = os.getenv("TERMINAL") or "x-terminal-emulator"
```

Reply:
that variable is often a `local` (it is on mine!) — you could just add an `opts.terminal` like you did for mixer!

Reply:
i'm thinking doing something like this
```lua
local popup_terminal = terminal or option.terminal or "urxvt"
```
 and then replace the terminal var in the other lines for popup_terminal

Reply:
`$TERMINAL` is not a standard Linux variable. (Also, `x-terminal-emulator` doesn't work on my machine, so I don't think it's standard either.)

> that way it still could be possible to not a terminal option when creating the widget in rc.lua

I know using the global is less work, but I think relying on a global may cause conflicts down the line — please use the option.

Reply:
at the end did 
```lua
local popup_terminal = options and options.terminal or terminal or "urxvt"
```
so options.terminal has priority, then the terminal from the global var on rc.lua and finally it will default to urxvt

Reply:
Sorry for the insistence, but please do not include the global; it is not a standard variable from the AwesomeWM environment. Even in the example `rc.lua` shipped by Awesome, `terminal` is passed around when it is intended to be used outside of that file:

https://github.com/awesomeWM/awesome/blob/master/awesomerc.lua#L77

As a general rule, I avoid having arbitrary globals cause side-effects on my modules.

Reply:
No problem, the insistence is not annoying at all, i will remove the global.

Reply:
okay, global removed 👍